runtime/fundamentals/modules.md

```diff
@@ -0,0 +1,450 @@
+
+```ts title="main.ts"
+import { camelCase } from "@luca/cases";
+
+camelCase("hello world"); // "helloWorld"
+```
+
+## Package Registries
+
+Deno recommends that you use [JSR](https://jsr.io/), the JavaScript registry, to
+publish and manage your modules for the best publishing experience. JSR offers
+automatic documentation generation, semver resolution and improved performance
+for TypeScript code. Deno also supports
+[other platforms](https://jsr.io/docs/other-registries) for publishing modules,
+such as npm, and JavaScript CDNs like deno.land/x, esm.h and unpkg.com
+
+By default when you use the CLI, the package will be resolved from JSR.
+
+```bash
+deno add @scopename/mypackage
+```
+
+The resulting import map contains the `jsr:` import specifier in the resulting
+`deno.json` file.
+
+```json
+{
+  "imports": {
+    "@scopename/mypackage": "jsr:@scopename/mypackage@^16.1.0"
+  }
+}
+```
+
+Deno supports the following registries:
+
+| Registry | Specifier |
+| -------- | --------- |
+| JSR      | jsr:      |
+| npm      | npm:      |
+
+To import a package from `npm` you need to prefix the package name with the
+`npm:` specifier:
+
+```bash
+$ deno add npm:uuid
+Add uuid - npm:uuid@^10.0.0
+```
+
+The resulting `deno.json` will contain the following import:
+
+```json
+{
+  "imports": {
+    "uuid": "npm:uuid@^10.0.0"
+  }
+}
+```
+
+You can then use the npm module by importing it by name from the import map.
+
+```ts title="mod.ts"
+import * as uuid from "uuid";
+
+console.log(uuid.v4());
+```
+
+## Importing modules from HTTP URLs
+
+Deno supports importing modules from HTTP URLs. Note that npm packages can be
+directly imported via the [`npm:` specifier](TODO:specifiers-link).
+
+```typescript
+import { render } from "https://esm.sh/preact";
+```
+
+You can also import modules from a URL by adding it to your `deno.json` import
+map:
+
+```json title="deno.json"
+{
+  "imports": {
+    "preact": "https://esm.sh/preact"
+  }
+}
+```
+
+URL imports should be used with caution, as they can introduce security risks.
+When importing modules from a URL, you are trusting the server to serve the
+correct code. If the server is compromised, it could serve malicious code to
+your application. For this reason, it is recommended to **use URL imports only
+from trusted sources**. They can also cause versioning issues if you import
+different versions in different files.
+
+## Package Versions
+
+It is possible to specify a version range for the package you are importing.
+This is done using the `@` symbol followed by a version range specifier, and
+follows the [semver](https://semver.org/) versioning scheme.
+
+```bash
+deno add @scopename/mypackage           # latest version
+deno add @scopename/mypackage@16.1.0    # exact version
+deno add @scopename/mypackage@^16.1.0   # latest patch version 16.x
+deno add @scopename/mypackage@~16.1.0   # latest version that doesn't increment the first non-zero portion
+```
```

Comment:
It's not just with `deno add`, also within specifiers.

The manual should explain the syntax of version ranges https://gist.github.com/jonlabelle/706b28d50ba75bf81d40782aa3c84b3e